src/domain/utils.ts

```diff
@@ -0,0 +1,24 @@
+import { ResultAsync } from "neverthrow"
+
+export const toTypedError = <ErrorType>({
+  unknownMessage,
+  _type,
+}): ErrorConverter<ErrorType> => {
+  return (error: any): ErrorWithMessage<ErrorType> => {
+    if (error.message && typeof error.message == "string") {
+      return { _type, message: error.message as string }
+    } else {
+      return { _type, message: unknownMessage }
+    }
+  }
+}
+
+export const unsafeThrowErrAsync = async <T, EType>(
```

Comment:
do we need this to make it work with current code? I thought the idea was to get rid of the exceptions.

Reply:
Yes we have 2 worlds the "old" try catch code and the "new" `Result` world. We are building the new world and integrating it into the old world. This is the translation back into the try/catch world. The goal is to never use this long term. Its just an indicator of how far the 'new' implementation has come.